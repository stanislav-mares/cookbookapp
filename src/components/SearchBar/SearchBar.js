import React from 'react';
import PropTypes from 'prop-types';

export default class SearchBar extends React.Component {

  constructor(props) {
    super(props);
  }

  handleChange = (event) => {
    this.props.handleChange(event);
  }

  handleClick = (event) => {
    this.props.toggleNewRecipeForm();
  }

  render() {
    return(
      <div>
        <input
          onChange={this.handleChange} value={this.props.value}
          type="text" name="searchBar" placeholder="Look for a recipe..."
        />
        <button onClick={this.handleClick}>+</button>
      </div>
    );
  }
}

SearchBar.propTypes = {
  handleChange : PropTypes.func.isRequired,
  value : PropTypes.string,
  toggleNewRecipeForm: PropTypes.func.isRequired
};
