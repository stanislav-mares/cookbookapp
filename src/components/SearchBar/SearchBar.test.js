import React from 'react';
import enzymeConfig from 'testConfig/enzymeConfig';
import { shallow } from 'enzyme';

import SearchBar from 'components/SearchBar/SearchBar';

describe('<SearchBar />', () => {
  it('Testing structure of the SearchBar component', () => {
    const component = shallow(
      <SearchBar
        handleChange={() => {}}
        toggleNewRecipeForm={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
