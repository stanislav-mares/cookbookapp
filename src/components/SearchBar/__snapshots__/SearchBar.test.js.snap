// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> Testing structure of the SearchBar component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleChange={[Function]}
    toggleNewRecipeForm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="searchBar"
          onChange={[Function]}
          placeholder="Look for a recipe..."
          type="text"
          value={undefined}
        />,
        <button
          onClick={[Function]}
        >
          +
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "name": "searchBar",
          "onChange": [Function],
          "placeholder": "Look for a recipe...",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="searchBar"
            onChange={[Function]}
            placeholder="Look for a recipe..."
            type="text"
            value={undefined}
          />,
          <button
            onClick={[Function]}
          >
            +
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "name": "searchBar",
            "onChange": [Function],
            "placeholder": "Look for a recipe...",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
